typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -0,0 +1,267 @@
+import { Mailbox } from '@hyperlane-xyz/core';
+import { Address, ProtocolType, rootLogger } from '@hyperlane-xyz/utils';
+
+import {
+  attachContractsMap,
+  serializeContractsMap,
+} from '../contracts/contracts.js';
+import { HyperlaneAddresses } from '../contracts/types.js';
+import { CoreConfig } from '../core/types.js';
+import { HyperlaneProxyFactoryDeployer } from '../deploy/HyperlaneProxyFactoryDeployer.js';
+import {
+  ProxyFactoryFactories,
+  proxyFactoryFactories,
+} from '../deploy/contracts.js';
+import { HyperlaneIsmFactory } from '../ism/HyperlaneIsmFactory.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { EthersV5Transaction } from '../providers/ProviderType.js';
+import { ChainNameOrId } from '../types.js';
+
+import {
+  HyperlaneModule,
+  HyperlaneModuleArgs,
+} from './AbstractHyperlaneModule.js';
+import { EvmIcaModule } from './EvmIcaModule.js';
+import { HyperlaneCoreDeployer } from './HyperlaneCoreDeployer.js';
+import { CoreFactories } from './contracts.js';
+
+// Partial CoreConfig because will be filled once .deploy() is called
+type DeployedAdresses = Partial<HyperlaneAddresses<CoreFactories>> & {
+  testRecipient?: Address;
+  timelockController?: Address;
+  interchainAccountRouter?: Address;
+  interchainAccountIsm?: Address;
+  deployedIsmFactoryFactories: HyperlaneAddresses<ProxyFactoryFactories>;
+};
```

Comment:
I would suggest having this type not be partial
iiuc the module constructor expects all these to be defined?

Reply:
Yeah that  makes more sense. updated